var/spack/repos/builtin/packages/openjdk/package.py

```diff
@@ -479,6 +487,37 @@ def install(self, spec, prefix):
         top_dir = "Contents/Home/" if platform.system() == "Darwin" else "."
         install_tree(top_dir, prefix)
 
+    @run_after("install")
+    def link_system_certs(self):
+        if self.spec.variants["certs"].value != "system":
+            return
```

Comment:
```suggestion
    @run_after("install")
    @when("certs=system")
    def link_system_certs(self):
```

We can clean this up by using an `@when()` decorator on the function to limit it to only running when the variant is set.

Reply:
one nice thing about using `@when` is that the decorated method will be filtered from the hash entirely for specs that don't match.  So if you make a change to this method it doesn't change the hash of specs where `certs` is not set to `system`.

Reply:
That's cool! I didn't know that